transfer/multichain_queue.go

```diff
@@ -0,0 +1,59 @@
+/*
+ * Copyright (C) 2021 The "MysteriumNetwork/payments" Authors.
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+package transfer
+
+import (
+	"errors"
+)
+
+// MultichainQueue is map of Queues which are able to
+// queue transaction execution funcs in parallel for each chain.
+type MultichainQueue struct {
+	qs map[int64]QueueIface
+}
+
+type QueueIface interface {
```

Comment:
Just queue is enough, Iface is superficial and just adds to noise.
```suggestion
type Queue interface {
```

Reply:
Can go with `Queue` as thats already a type name, but you can get a bit more creative. In general we want to avoid `types` in the name for a type. If you make a map of strings you dont want to call it `StrMap` etc.

Reply:
Changed it to Enqueuer